### PR TITLE
markers symbols from mapbox

### PR DESCRIPTION
**Pull request by stebogit**
Looks like I found a way to render the markers however there are a couple of issues I'm not sure how to solve:
- I used the code found [here](http://bl.ocks.org/tmcw/3861338), which is also used on [geojson.io](http://geojson.io/), to define the icon url without using a token, whereas [the original code](https://github.com/mapbox/mapbox.js/blob/publisher-production/src/marker.js#L26) requires one. Is this legit?
- I could not find a way to define `pointToLayer` only if `useSimpleStyle = true`; maybe extending the GeoJSON class? But I wasn't able to do that 😞 
Any idea?

Screen Shot 2020-03-29 at 2 59 42 AM

### CODE REVIEW

File: src/simplestyle.js [hunk 1]
```diff
@@ -38,27 +38,29 @@ function remap(a) {
...
         size = fp['marker-size'] || 'medium',
+        // see valid symbols https://labs.mapbox.com/maki-icons/
+        symbol = ('marker-symbol' in fp && fp['marker-symbol'] !== '') ? `-${  fp['marker-symbol']}` : '',
         color = (fp['marker-color'] || '7e7e7e').replace('#', '');
 
-    const svg = `<svg width='354' height='458' xmlns='http://www.w3.org/2000/svg'><path d='M177 0C79 0 0 79 0 177c0 95 163 270 170 277a9 9 0 0013 0c7-7 170-182 170-277C353 79 274 0 177 0z' fill='#${color}'/></svg>`
     return L.icon({
-        iconUrl: encodeURI(`data:image/svg+xml,${svg}`).replace('#', '%23'),
+        // from http://bl.ocks.org/tmcw/3861338
+        iconUrl: `//a.tiles.mapbox.com/v3/marker/pin-${
```
Comment (stebogit) [inline src/simplestyle.js]: I believe, for some reason, this API does not require a token. As I mentioned it is the same used in geojson.io, so I figure it's ok here?

File: package.json [hunk 2]
```diff
@@ -12,7 +12,8 @@
   },
   "scripts": {
     "build": "rollup -c",
-    "dev": "rollup -c -w"
+    "dev": "rollup -c -w",
+    "lint": "eslint src/ --fix"
```
Comment (stebogit) [inline package.json]: even running eslint there are still a bunch of unwanted white spaces, sorry

**Comment by rowanwins** [comment]
The downside I see of the mapbox icons is that it requires calling out to an external site which makes this package reliant on that always being available which may not always be desired (eg in a network restricted environment).

I'm wondering if we can perhaps pass another option like `useMakiMarkers` which determines whether you get icon support or not...

Another option again might be to actually pull in the maki repo and the SVG's
https://github.com/mapbox/maki

Choices choices - I'll have a ponder...

**Comment by stebogit** [comment]
What about extending the `Icon` class to add a fallback image to the resulting `img` element in case the url did not resolved?
https://stackoverflow.com/a/9891041/4530144

Anyway, I like the additional `useMakiMarkers` (default `true`).

**Pull request closed without merging**